core/src/test/scala/pickling/run/runtime-instantiation-static-picklers.scala

```diff
@@ -0,0 +1,219 @@
+package scala.pickling.runtime
+
+import org.scalatest.FunSuite
+import scala.pickling._, scala.pickling.Defaults._, json._
+
+case class Container[T](value: T)
+
+object RuntimeHelper {
+  /* Instantiate a class for an already generated `Unpickler` */
+  def getInstance[T](className: String): T = {
+    try {
+      val clazz = Class.forName(className)
+      println(s"Create instance of $clazz")
+      clazz.newInstance().asInstanceOf[T]
+    } catch {
+      case ex: Throwable =>
+        scala.concurrent.util.Unsafe.instance.allocateInstance(
+          Class.forName(className)
+        ).asInstanceOf[T]
+    }
+  }
+}
+
+class StaticJavaRuntime extends FunSuite {
+
+  implicit val staticOnly = static.StaticOnly
+
+  test("runtime instantiated pickler should work") {
+
+    val c = Container(List(1,2,3,4,5))
+    val containerPickler = implicitly[Pickler[Container[List[Int]]]]
+    val pickled = c.pickle.value
+
+    val picklerClassName = containerPickler.getClass.getName
+    val runtimePickler = RuntimeHelper.getInstance[Pickler[Container[Any]]](picklerClassName)
```

Comment:
Oh, wait, I think I get now what you mean... You're actually serializing the bytes that represent the class (sorry, I miss that detail). That's a great hack, darn it. However, this solution would be the least one I'd like to use since 1. it considerably increases the size of the exchanged messages among the nodes of a distributed system, and 2. It's a very hacky solution that creates an important security hole because you're loading picklers/unpicklers on the fly.
  
Wouldn't it be possible to have a sort of dictionary of class names and custom picklers/unpicklers that is populated when the scala pickling initializers are executed?